### PR TITLE
Register squashfs filesystem

### PR DESCRIPTION
**Pull request by JSCU-CNI**
This PR registers squashfs as a filesystem for dissect.target.

### CODE REVIEW

**Review by pyrco — CHANGES_REQUESTED** [review]
Thanks for this missing squashfs piece!

In general it looks good. However `is_dir()`, `if_file()` and `stat()`  these days have an extra `follow_symlinks` parameter (to ensure support for Python >= 3.11.4). Could you implement those as well? The implementations in `extfs.py` can be used as a good example.

**Comment by JSCU-CNI** [comment]
Thanks for the suggestion. [Added](https://github.com/fox-it/dissect.target/pull/324/commits/329dfcfb3f21b30b91201ec5144a83ecf0790b15) updated `is_dir()`, `is_file()`, `is_symlink()` and `stat()` functions. Also removed unnecessary `_resolve()` definition as it's identical to the inherited `FilesystemEntry` class and tweaked the returned `st_info` inside `stat()`.

File: dissect/target/filesystems/squashfs.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    __fstype__ = "squashfs"
+
+    def __init__(self, fh: BinaryIO, *args, **kwargs):
+        super().__init__(fh, *args, **kwargs)
+        self.squashfs = SquashFS(fh)
+
+    @staticmethod
+    def _detect(fh: BinaryIO) -> bool:
+        sb = c_squashfs.c_squashfs.squashfs_super_block(fh)
+        return sb.s_magic == c_squashfs.c_squashfs.SQUASHFS_MAGIC
```
Comment (Schamper) [inline dissect/target/filesystems/squashfs.py:27]:
```suggestion
        return int.from_bytes(fh.read(4), "little") == c_squashfs.c_squashfs.SQUASHFS_MAGIC
```

File: dissect/target/filesystems/squashfs.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+    def lstat(self) -> fsutil.stat_result:
+        node = self.entry
+
+        st_info = fsutil.stat_result(
+            [
+                node.mode,
+                node.inode_number,
+                id(self.fs),
+                0,  # nlink
```
Comment (Schamper) [inline dissect/target/filesystems/squashfs.py:111]:
```suggestion
                getattr(node.header, "nlink", 0),
```
?

File: dissect/target/filesystems/squashfs.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                node.size,
+                0,  # atime
+                node.mtime.timestamp(),
+                0,  # ctime
+            ]
+        )
+
+        if "nlink" in dir(node):
+            st_info.st_nlink = node.nlink
+
```
Comment (Schamper) [inline dissect/target/filesystems/squashfs.py:123]:
```suggestion
```

**Review by Schamper — CHANGES_REQUESTED** [review]

**Review by pyrco — APPROVED** [review]

**Review by Schamper — APPROVED** [review]

**Pull request merged**